src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -68,6 +69,24 @@ public JsonArray(List list) {
   }
 
   /**
+   * Create an instance from a Buffer of JSON.
+   *
+   * @param buf  the buffer of JSON.
+   */
+  public JsonArray(Buffer buf) {
+    fromBuffer(buf);
+  }
+
+  /**
+   * Create an instance from a stream of JSON.
+   *
+   * @param stream  the stream of JSON.
+   */
+  public JsonArray(InputStream stream) {
```

Comment:
Not totally sure about this, since it would allow the code to be abused with blocking IO as the InputStream is blocking by nature